docs/manage-fileset-metadata-using-gravitino.md

```diff
@@ -429,34 +465,198 @@ catalog.as_fileset_catalog().create_fileset(ident=NameIdentifier.of("test_schema
+```shell
+# create a catalog first
+curl -X POST -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{
+  "name": "test_catalog",
+  "type": "FILESET",
+  "comment": "comment",
+  "provider": "hadoop",
+  "properties": {
+    "filesystem-providers": "builtin-local,builtin-hdfs,s3,gcs",
+    "location-l1": "file:///{{catalog}}/{{schema}}/workspace_{{project}}/{{user}}",
+    "location-l2": "hdfs:///{{catalog}}/{{schema}}/workspace_{{project}}/{{user}}"
+  }
+}' http://localhost:8090/api/metalakes/metalake/catalogs
 
-3. When the catalog property `location` isn't specified but the schema property `location` is specified,
-   the storage location is:
-   1. `schema location/fileset name` if `schema location` does not contain any placeholder.
-   2. `schema location` - placeholders in the schema location will be replaced by the corresponding fileset property value.
-   
-4. When both the catalog property `location` and the schema property `location` are specified, the storage
-   location is:
-   1. `schema location/fileset name` if `schema location` does not contain any placeholder.
-   2. `schema location` - placeholders in the schema location will be replaced by the corresponding fileset property value.
+# create a schema under the catalog
+curl -X POST -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{
+  "name": "test_schema",
+  "comment": "comment",
+  "properties": {
+    "location-l3": "s3a://myBucket/{{catalog}}/{{schema}}/workspace_{{project}}/{{user}}"
+  }
+}' http://localhost:8090/api/metalakes/metalake/catalogs/test_catalog/schemas
 
-5. When both the catalog property `location` and schema property `location` isn't specified, the user
-   should specify the `storageLocation` in the fileset creation.
+# create a fileset by placeholders
+curl -X POST -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{
+  "name": "example_fileset",
+  "comment": "This is an example fileset",
+  "type": "MANAGED",
+  "storageLocations": {
+    "l4": "gs://myBucket/{{catalog}}/{{schema}}/workspace_{{project}}/{{user}}"
+  },
+  "properties": {
+    "placeholder-project": "test_project",
+    "placeholder-user": "test_user",
+    "default-location-name": "l1"
+  }
+}' http://localhost:8090/api/metalakes/metalake/catalogs/test_catalog/schemas/test_schema/filesets
 
-For `EXTERNAL` fileset, users should specify `storageLocation` during the fileset creation,
-otherwise, Gravitino will throw an exception. If the `storageLocation` contains placeholders, the
-placeholder will be replaced by the corresponding fileset property value.
+# the fileset will be created with 4 storage locations:
+{
+  "name": "example_fileset",
+  "comment": "This is an example fileset",
+  "type": "MANAGED",
+  "storageLocation": null,
+  "storageLocations": {
```

Comment:
`storageLocation` is for backward compatibility?

Why do we have both singular and plural forms of the same property?

Can we unify this into just a `storageLocations` map?

Reply:
Another question ...
Is `storageLocation` a required property?
Why do we need to explicitly set a property that might be defaulted to null?

Reply:
> `storageLocation` is for backward compatibility?
> 
> Why do we have both singular and plural forms of the same property?
> 
> Can we unify this into just a `storageLocations` map?

yeah, `storageLocation` is for backward compatibility.

Reply:
> Is `storageLocation` a required property?

No.

> Why do we need to explicitly set a property that might be defaulted to null?

This is not a creation request, but an example of a returned result.

Reply:
Makes sense.